src/Data/Singletons/Prelude/Show.hs

```diff
@@ -0,0 +1,229 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DefaultSignatures #-}
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeInType #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{-# LANGUAGE TypeApplications #-}
+
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  Data.Singletons.Prelude.Show
+-- Copyright   :  (C) 2013 Richard Eisenberg
+-- License     :  BSD-style (see LICENSE)
+-- Maintainer  :  Richard Eisenberg (rae@cs.brynmawr.edu)
+-- Stability   :  experimental
+-- Portability :  non-portable
+--
+-- Defines the SShow singleton version of the Show type class.
+--
+-----------------------------------------------------------------------------
+
+module Data.Singletons.Prelude.Show (
+  PShow(..), SShow(..), SymbolS, SChar,
+
+  ShowsPrecSym0, ShowsPrecSym1, ShowsPrecSym2, ShowsPrecSym3,
+  Show'Sym0, Show'Sym1,
+  ShowListSym0, ShowListSym1, ShowListSym2,
+
+  (:<>), (%:<>),
+  (:<>$), (:<>$$), (:<>$$$),
+
+  Shows, sShows,
+  ShowsSym0, ShowsSym1, ShowsSym2,
+
+  ShowListWith, sShowListWith,
+  ShowListWithSym0, ShowListWithSym1, ShowListWithSym2, ShowListWithSym3,
+
+  ShowChar, sShowChar,
+  ShowCharSym0, ShowCharSym1, ShowCharSym2,
+
+  ShowString, sShowString,
+  ShowStringSym0, ShowStringSym1, ShowStringSym2,
+
+  ShowParen, sShowParen,
+  ShowParenSym0, ShowParenSym1, ShowParenSym2,
+
+  ShowSpace, sShowSpace,
+  ShowSpaceSym0, ShowSpaceSym1,
+
+  ShowCommaSpace, sShowCommaSpace,
+  ShowCommaSpaceSym0, ShowCommaSpaceSym1,
+
+  AppPrec, sAppPrec, AppPrecSym0,
+
+  AppPrec1, sAppPrec1, AppPrec1Sym0
+  ) where
+
+import           Data.Kind (Type)
+import           Data.List.NonEmpty (NonEmpty)
+import           Data.Monoid ((<>))
+import           Data.Singletons.Prelude.Base
+import           Data.Singletons.Prelude.Instances
+import           Data.Singletons.Prelude.List
+import           Data.Singletons.Prelude.Ord
+import           Data.Singletons.Single
+import           Data.Singletons.TH
+import           Data.Singletons.TypeLits
+import qualified Data.Text as T
+
+import           GHC.TypeLits
+
+import qualified Prelude as P
+import           Prelude hiding (Show(..))
+
+import           Unsafe.Coerce (unsafeCoerce)
+
+-- | The promoted analogue of '(<>)' for 'Symbol's. This uses the special
```

Comment:
That's a good point. What we should _really_ be doing is defining a `Monoid` instance for `Symbols` first, and then use `Data.Monoid.<>` instead of defining it here. But we don't have a `Monoid` class in `singletons` yet, so I went with the simpler approach of rolling my own.